### ISSUE TITLE
Use Case: replace a failed disk

### ISSUE THREAD

**Issue by portante** [heketi/heketi#633]
What is the story given for how to replace a failed disk in operation?

What I'd like to be able to say is something like (pseudo command/english syntax):
```
heketi replace disk 'disk15' on host1.example.com with 'disk76' on hostX.example.com
```

And then Heketi would take care of all the necessary steps to remove the bricks made from the failed disks with identical bricks constructed from the new disk.

Ideally, Heketi would warn the user of the replacement disk is too small to handle the replacement load.

Also, as noted above, it might be the case that the replacement disk lives on the same host, or on a different host.

What do folks think about this?

**Comment by obnoxxx** [comment]
Related:
https://github.com/heketi/heketi/issues/161

Currently the approach of heketi would be to automatically chose new disks when a disk is flagged as failed. But since this not yet fully implemented, we can still talk about it...

**Comment by obnoxxx** [comment]
Also see the involved documentation for replacing bricks in a volume:

https://gluster.readthedocs.io/en/latest/Administrator%20Guide/Managing%20Volumes/#replace-brick

**Comment by think01** [comment]
@obnoxxx than is the procedure to operate on Gluster. But if I do that, doesn't Heketi get desynch from the cluster' status? I think I should replicate the substitution operation at the Heketi level but I cannot figure out how to accomplish that. Is it implemented someway?

**Comment by boxdaxter** [comment]
Hi, 
@lpabon, @obnoxxx   

We are implementing Heketi as SWARM persistent storage. We found the same limitation than is described in this post. How we should manage a brick, device or node removal in case os fail? Our Volumes are big, >= 200GB and more than 20 bricks/volume, it's not possible to fix it by hand.

As @think01 says, if we fix the gluster cluster by hand following the RedHat community procedure, the heketi's bbdd will discover the changes on the volumes, or it will be desync with the real state of the volume?

Regards.
Thanks.

**Comment by lpabon** [comment]
Hi @portante, @boxdaxter , @think01 ,
  my next goal is to work on #161 which is greatly needed to fix the issues above.  Stay tuned.

Currently the _only_ way to fix this is to create a New volume, then copy the data, then delete the old.  Yes, I know that is *terrible*, but it works. :-D

**Comment by portante** [comment]
Thanks @lpabon!

@t0ffel, this is what we'll have to do for now...

**Comment by obnoxxx** [comment]
@portante @lpabon see my comment on #161 : @raghavendra-talur and @MohamedAshiqrh are already working on "replace device" (trying to avoid duplication of effort here...).

**Comment by obnoxxx** [comment]
@think01 Right, I did not mean to say to use gluster commands for replace brick (etc) on a heketi-managed cluster, but quoted the documentation to indicate the difficulties involved in getting it right. And pointing out what needs to be done by heketi.

**Comment by MohamedAshiqrh** [comment]
https://github.com/heketi/heketi/pull/676

Will update the PR.

**Comment by think01** [comment]
@obnoxxx I see now, thanks for the reply. We thought it was some sort of workaround so we tried (and) failed hard to apply it.

**Comment by raghavendra-talur** [comment]
Can we consider that #710 closes this?

**Comment by lpabon** [comment]
Closed by #710
